src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineViewModel_Utilities.cs

```diff
@@ -120,7 +120,7 @@ public static ImmutableArray<DocumentSymbolData> CreateDocumentSymbolData(JToken
         while (currentStart < allSymbols.Length)
             finalResult.Add(NestDescendantSymbols(allSymbols, currentStart, out currentStart));
 
-        return finalResult.ToImmutable();
+        return finalResult.ToImmutableAndClear();
```

Comment:
>finalResult [](http://example.com/codeflow?start=15&length=11)

nit: another

Reply:
Not enough coffee yet, which conditionals?

Reply:
```c#
        var currentStart = 0;
        while (currentStart < allSymbols.Length)
            finalResult.Add(NestDescendantSymbols(allSymbols, currentStart, out currentStart));
```

The loop depends on currentStart.  but current start is mutated by the body.  so i don't think we know how many elements we actually have.

Reply:
Gotcha, missed seeing that a couple times